src/terms/modified_kinetic_blow_up_function.jl

```diff
@@ -0,0 +1,92 @@
+# Implementation of the blow-up function used to compute modified kinetic energies.
+# For a given interval [x_1,x_2] ⊂ [0,1) and a blow-up order p>0, the blow-up function Gp is
+# composed of three parts:
+#
+#  - g1(x) = x^2 defined on [0,x_1),
+#  - g3(x, p) ≥ x^2 ∀p, defined on [x_2,1), that blows-up at x=1 with blow-up rate |⋅|^{-p},
+#  - g2, a C^2 interpolation between g1 and g3_p on [x_1,x_2].
+#
+# In practice we define:
+#
+# g3(x,p) = g3(x, a, p) = C(a,p)/(1-x)^p, where C(a,p) = (3/2)*(a^2)*(1-a)^(p)
+#
+# and where the parameter a is optimized so that ||Gp - x^2||_{L^1((0,1))} is minimized.
+
```

Comment:
Looks like it's something that was used to generate the blowup function and does not need to be in DFTK? Just hardcode the coefficients. Eg you can have in kinetic.jl things like blowup_function_CHV (for cancès hassan vidal), and blowup_function_abinit